shinydashboard/lantern/app.R

```diff
@@ -0,0 +1,231 @@
+
+# Get the list of most recent HTTP responses when requesting the capability statement from the 
+# fhir_endpoints
+response_tally <- list()
+response_tally$http_200 <- curr_http_response_tally %>% filter(http_response==200) %>% pull(n)
+response_tally$http_404 <- curr_http_response_tally %>% filter(http_response==404) %>% pull(n)
+response_tally$http_503 <- curr_http_response_tally %>% filter(http_response==503) %>% pull(n)
+
+http_pct <- as_tibble(https_response_values %>% 
+                          mutate(code=as.character(value)) %>%
+                          group_by(labels_id,code,value) %>% 
+                          summarise(Percentage=n()) %>% 
+                          group_by(labels_id) %>% 
+                          mutate(Percentage=Percentage/sum(Percentage,na.rm = TRUE)*100)
+)
+
+http_pctf <- http_pct %>% filter(value != 200) %>% mutate(name=as.factor(labels_id), code=as.factor(code)) 
+
+http_summary <- http_pct %>%
+    left_join(http_response_code_tbl, by=c("code" = "code_chr")) %>%
+    select(code,label) %>%
+    group_by("HTTP Response" = code,"Status"=label) %>%
+    summarise(Count=n()) 
+
+fhir_version_tbl <- as_tibble(tbl(con,sql("select id,url,vendor,capability_statement->>'fhirVersion' as FHIR from fhir_endpoints where capability_statement->>'fhirVersion' IS NOT NULL")))
+fhir_version_list <- as.list(fhir_version_tbl %>% distinct("FHIR Version"=fhir))
+fhir_version_vendor_count <- fhir_version_tbl %>%
+    mutate(vendor = na_if(vendor,"")) %>%
+    tidyr::replace_na(list(vendor="Unknown")) %>%
+    group_by(vendor,fhir) %>%
+    tally() %>%
+    select(Vendor=vendor,"FHIR Version"=fhir,"Count"=n)
+
+
+vendor_list <- as.list(as_tibble(fhir_endpoints %>% distinct(vendor)) %>% mutate(vendor = na_if(vendor,"")) %>% tidyr::replace_na(list(vendor="Unknown")) %>% pull(vendor))
+
+foo <- tbl(con,sql("SELECT floor(extract(epoch from metrics_values.time)/240)*240 AS time, AVG(metrics_values.value) 
+FROM metrics_labels, metrics_values
+WHERE metrics_labels.metric_name = 'AllEndpoints_http_response_time' 
+AND metrics_labels.id = metrics_values.labels_id
+AND metrics_values.time BETWEEN '2020-04-07T22:40:39.024Z' AND '2020-04-14T22:40:39.024Z'
+GROUP BY time
+ORDER BY time "))
+ff <- as_tibble(foo)
+
+# Define the UI components
+ui <- dashboardPage(
+    dashboardHeader(
+        title = "Lantern Dashboard",
+        titleWidth = 200
+    ),
+    # Sidebar with a slider input for number of bins 
+    dashboardSidebar(
+        sidebarMenu(
+            menuItem("Dashboard", tabName = "dashboard", icon = icon("dashboard"),selected=TRUE),
+            menuItem("Availability", icon = icon("th"), tabName = "availability", badgeLabel = "new",
+                     badgeColor = "green"
+            ),
+            menuItem("Performance", icon = icon("bar-chart-o"),
+                     menuSubItem("Mean Response Time", tabName = "subitem1"),
+                     menuSubItem("Performance sub-item", tabName = "subitem2")
```

Comment:
this doesn't look like it's being used?

Reply:
The Performance sub-item is placeholder. Will change as additional vis. pages are added.